WcaOnRails/app/models/competition.rb

```diff
@@ -30,6 +32,29 @@ class Competition < ActiveRecord::Base
            with_model_currency: :currency_code
 
   scope :not_over, -> { where("CAST(CONCAT(endYear,'-',endMonth,'-',endDay) as Datetime) >= ?", Date.today) }
+  scope :belongs_to_region, lambda { |region_id|
+    joins(:country).where(
+      "countryId = :region_id OR Countries.continentId = :region_id", region_id: region_id
+    )
+  }
+  scope :contains, lambda { |search_term|
+    joins(:delegates).where(
+      "Competitions.name like :search_term or
+      Competitions.cityName like :search_term or
+      users.name like :search_term or
+      Competitions.venue like :search_term or
+      Competitions.cellName like :search_term or
+      Competitions.countryId like :search_term or
+      MONTHNAME(STR_TO_DATE(Competitions.month, '%m')) like :search_term",
+      search_term: "%#{search_term}%"
```

Comment:
I was discussing with Jeremy and I'm not sure if people use the month or venue search. I don't think they make much sense. Same for country, since we have an explicit country/region dropdown. I think we could remove this from here. Name and city should be enough. 
About delegate, I use the search currently, but could live with going through /admin/delegates to see someone's history. We could then use a different param (delegate_id) instead of passing it through search param. See #1035